applications/asset_tracker_v2/src/cloud/cloud_codec/aws_iot_codec.c

```diff
@@ -20,6 +20,35 @@
 #include <logging/log.h>
 LOG_MODULE_REGISTER(cloud_codec, CONFIG_CLOUD_CODEC_LOG_LEVEL);
 
+/* Function that checks the version number of the incoming message and determines if it has already
+ * been handled. Receiving duplicate messages can often occur upon retransmissions from the AWS IoT
+ * broker due to unACKed MQTT QoS 1 messages and unACKed TCP packets. Messages from AWS IoT
+ * shadow topics contains an incrementing version number.
+ */
+static bool has_message_been_handled(cJSON *root_obj)
+{
+	cJSON *version_obj;
+	static int version_prev;
+	bool retval = false;
+
+	version_obj = cJSON_GetObjectItem(root_obj, DATA_VERSION);
+	if (version_obj == NULL) {
+		/* No version number present in message. */
+		return false;
+	}
+
+	/* If the incoming version number is lower than or equal to the previous,
+	 * the incoming message is concidered a retransmitted message and will be ignored.
```

Comment:
```suggestion
	 * the incoming message is considered a re-transmitted message and will be ignored.
```

Reply:
Isn't ```retransmitted``` and ```re-transmitted``` equally valid? ```retransmitted``` is often enforced throughout NCS. Not that it matters.

Reply:
Both are fine.